apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -309,3 +321,106 @@ class FrameworkError {
   message: string;
   name: string;
 }
+
+function sanitizeControlValuesByOutputSchema(
+  controlValues: Record<string, unknown>,
+  type: StepTypeEnum
+): Record<string, unknown> {
+  const outputSchema = channelStepSchemas[type].output || actionStepSchemas[type].output;
+
+  if (!outputSchema || !controlValues) {
```

Comment:
We should throw an error at this point. The inability to identify an `outputSchema` from the framework indicates an invalid workflow definition.

Reply:
agreed, but I tried to handle it as gracefully as possible. maybe we could log and return the controlValues instead? are you sure we need to throw an error here?

Reply:
Yes.